pyQuARC/code/scheduler.py

```diff
@@ -27,13 +27,13 @@ def get_all_dependencies(self, rule_id, check, field_dict=None):
         dependencies_from_fields = []
 
         if field_dict:
-            dependencies_from_fields = field_dict.get("dependencies", [])
-        else:
-            rule = self.rule_mapping.get(rule_id)
-            if field_objects := rule.get("fields_to_apply").get(self.metadata_format):
-                for field_object in field_objects:
-                    if field_dependencies := field_object.get("dependencies"):
-                        dependencies_from_fields.extend(field_dependencies)
+            return field_dict.get("dependencies", [])
+
+        rule = self.rule_mapping.get(rule_id)
+        if field_objects := rule.get("fields_to_apply").get(self.metadata_format):
+            for field_object in field_objects:
+                if field_dependencies := field_object.get("dependencies"):
+                    dependencies_from_fields.extend(field_dependencies)
```

Comment:
Do we not need to go through these steps even when the field_dict is present?

Reply:
no, field_dict contains the specific dependencies that relate to the given field, so we need only that
`field_dict` isn't always available
so we have to search through all the dependencies when it's not there, used when we need to order the calls to the rules based on the dependencies